src/settings.rs

```diff
@@ -348,6 +526,90 @@ impl ContextSettingExt for Context {
+        if let Some(previous_mod_itinerary) = container.modified_itineraries.get(&person_id) {
+            container
+                .remove_member_from_itinerary_settings(person_id, previous_mod_itinerary.clone());
+        }
+
+        container.modified_itineraries.remove(&person_id);
+
+        // Get people back to default itinerary, if there's one
+        match container.itineraries.get(&person_id) {
+            None => {
+                return Err(IxaError::from(
+                    "Can't modify itinerary if there isn't one present",
+                ))
+            }
+            Some(default_itinerary) => {
+                for itinerary_entry in default_itinerary {
+                    if !container
+                        .setting_types
+                        .contains_key(&itinerary_entry.setting_type)
+                    {
+                        return Err(IxaError::from(
+                            "Itinerary entry setting type not registered",
+                        ));
+                    }
+
+                    //container.add_member_to_itinerary_setting(person_id, itinerary_entry);
+                    container
+                        .members
+                        .entry(itinerary_entry.setting_type)
+                        .or_default()
+                        .entry(itinerary_entry.setting_id)
+                        .or_default()
+                        .push(person_id);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn modify_itinerary(
+        &mut self,
+        person_id: PersonId,
+        itinerary_modifier: ItineraryModifiers,
+    ) -> Result<(), IxaError> {
+        match itinerary_modifier {
+            ItineraryModifiers::ReplaceWith { itinerary } => {
+                println!("ItineraryModifier::Replace --  {itinerary:?}");
+                self.add_modified_itinerary(person_id, itinerary)
+            }
+            ItineraryModifiers::Include { setting } => {
+                println!("ItineraryModifier::Include -- {:?}", setting.get_type_id());
+                self.limit_itinerary_by_setting_type(person_id, setting)
+            }
+            ItineraryModifiers::Exclude { setting } => {
+                println!("ItineraryModifier::Exclude -- {:?}", setting.get_type_id());
+                self.exclude_setting_from_itinerary(person_id, setting)
+            }
+        }
+    }
+
+    fn get_setting_id<T: SettingType>(
+        &mut self,
+        person_id: PersonId,
+        setting_type: T,
+    ) -> Vec<usize> {
+        let container = self.get_data_container_mut(SettingDataPlugin);
+        match container.itineraries.get(&person_id) {
+            None => Vec::new(),
+            Some(itineraries) => {
+                let mut setting_id_vec = Vec::new();
+                for entry in itineraries {
+                    if entry.setting_type == setting_type.get_type_id() {
+                        setting_id_vec.push(entry.setting_id);
```

Comment:
How does this work?

```suggestion
                        setting_id_vec.push(SettingId::new(entry.setting_type, entry.setting_id));
```

You may need to make `Copy` a supertrait of `SettingType`. I will mess around on locally too if I get a chance. This enables us to change the signature of this method to be `Vec<SettingId<T>>`.